c_level_db.go

```diff
@@ -274,8 +274,8 @@ func (itr cLevelDBIterator) Valid() bool {
 		return false
 	}
 
-	// Panic on DB error.  No way to recover.
-	itr.assertNoError()
+	// checks no error is present
+	itr.checkNoError()
```

Comment:
return false if err?

Reply:
it may make sense to have valid() return an error instead of panicing like it is in other places, let me know if you agree?

Reply:
I think we should 

1) leave assertIsValid and let it panic if method is called while Valid() is false
2) add new func Error() error
```go
func (itr cLevelDBIterator) Error() error {
return itr.source.Error()
}
```
3) leave assertNoError  and let it panic if method is called while Error() is not nil

Reply:
4. modify Key, Value, Next to not return an error

Reply:
made the changes as suggested.